### PR TITLE
Release 2.0

### PR DESCRIPTION
**Pull request by augustocdias**
Merge the 2.0 branch into release.

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

_Briefly describe the issue you have solved or implemented with this pull request. If the PR contains multiple issues, use a bullet list._

### Causes (Optional)

_Briefly describe the causes behind the issues. This could be helpful to understand the adopted solutions behind some nasty bugs or complex issues._

### Solutions

_Briefly describe the solutions you have implemented for the issues explained above._

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Cargo.lock
Comment (SimonThormeyer) [inline Cargo.lock:1]: Why are we removing `Cargo.lock`?

File: .cargo/audit.toml
Comment (SimonThormeyer) [inline .cargo/audit.toml:1]: If we got rid of `libsodium` and `sodiumoxide`, we can remove this I guess?

File: .cargo/audit.toml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+[advisories]
+ignore = ["RUSTSEC-2021-0137"]
```
Comment (SimonThormeyer) [inline .cargo/audit.toml:2]: If we got rid of libsodium and sodiumoxide, we can remove this I guess?

File: .gitignore [hunk 2]
```diff
@@ -1,4 +1,5 @@
-/target
+target
+Cargo.lock
```
Comment (SimonThormeyer) [inline .gitignore]: Do we really not want to track `Cargo.lock`?

File: Cargo.toml [hunk 3]
```diff
@@ -1,20 +1,73 @@
 [package]
-
-name         = "proteus"
-version      = "1.0.4"
+name         = "proteus-wasm"
```
Comment (SimonThormeyer) [inline Cargo.toml:2]: Is this just gonna be used in wasm builds?

File: Cargo.toml [hunk 4]
```diff
@@ -1,20 +1,73 @@
...
+    "crates/proteus-traits",
+    "crates/cbor-codec",
+]
+
+[patch.'https://github.com/wireapp/proteus'.proteus]
+package = "proteus"
+# Keep the extra / in the url, it prevents this error:
+# `patch for `proteus` in `https://github.com/wireapp/proteus` points to the same source, but patches must point to different sources`
+git = "https://github.com/wireapp//proteus"
+branch = "otak/fix-1.0.3"
```
Comment (SimonThormeyer) [inline Cargo.toml:22]: Is this intended? if yes, doesn't this defeat the purpose of having a single source (branch) of truth?

Comment (augustocdias) [reply Cargo.toml:22]: This is to test with older version to ensure it is retro-compatible

File: Cargo.toml (hunk 3 above)
Comment (augustocdias) [reply Cargo.toml:2]: No... This a version that works with wasm and needed a new name to not conflict with the old version when both are present in the dependency tree.

File: .cargo/audit.toml (hunk 1 above)
Comment (augustocdias) [reply .cargo/audit.toml:2]: we still have proteus 1.x in the tree, so this is still required.

**Review by SimonThormeyer — APPROVED** [review]

**Pull request merged**